src/main/java/org/dependencytrack/search/FuzzyVulnerableSoftwareSearchMananger.java

```diff
@@ -0,0 +1,268 @@
+package org.dependencytrack.search;
+
+import alpine.common.logging.Logger;
+import alpine.notification.Notification;
+import alpine.notification.NotificationLevel;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.CorruptIndexException;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.VulnerableSoftware;
+import org.dependencytrack.notification.NotificationConstants;
+import org.dependencytrack.notification.NotificationGroup;
+import org.dependencytrack.notification.NotificationScope;
+import org.dependencytrack.persistence.QueryManager;
+import us.springett.parsers.cpe.CpeParser;
+import us.springett.parsers.cpe.exceptions.CpeParsingException;
+import us.springett.parsers.cpe.exceptions.CpeValidationException;
+import us.springett.parsers.cpe.values.Part;
+
+import java.io.IOException;
+import java.util.*;
+
+public class FuzzyVulnerableSoftwareSearchMananger {
+
+    private static final Logger LOGGER = Logger.getLogger(FuzzyVulnerableSoftwareSearchMananger.class);
+    private static final Set<String> DO_NOT_FUZZ = Collections.unmodifiableSet(Set.of("util", "utils", "url", "xml"));
+
+    private final boolean excludeComponentsWithPurl;
+    private final Set<String> SKIP_LUCENE_FUZZING_FOR_TYPE = Collections.unmodifiableSet(Sets.newHashSet("golang"));
+    public FuzzyVulnerableSoftwareSearchMananger(boolean excludeComponentsWithPurl) {
+        this.excludeComponentsWithPurl = excludeComponentsWithPurl;
+    }
+
+    private class SearchTerm {
```

Comment:
This class should be static.

```suggestion
    private static class SearchTerm {
```